### PR TITLE
Fix #813, we can now have multiple docs or SPs open

### LINKED ISSUE #813: [Suggestion]Keep consistent when adding a collection between VS Code and portal
Repro steps:
1. Right click on a SQL account -> Select 'Open In Portal' -> Select 'Add Collection'.
2. Drill down from the same SQL account in Step1 ->  Create a new database -> Right click on this new database -> Select 'Create Collection'.

Actual:
In VS Code, it prompts user to enter a collection name for this new collection.
In portal, it prompts user to enter a collection id for this new collection.

Suggestion:
Keep consistent when adding a collection between VS Code and portal.
capture3

### PR DESCRIPTION
**Pull request by tec-goblin**
Before the name was hardcoded to identify whether we can use the upload command.
Now it's based on a pattern in the name. The upload command still works both as an icon and from the contextual menu.

### CODE REVIEW

**Comment by tec-goblin** [comment]
Hi, can you explain me the steps to reproduce this?

**Comment by PrashanthCorp** [comment]
If I understand correctly, `/(.*cosmos-doc[.]json)/` would match cosmos-doc.json, but not cosmos-document.json. Some users might save the documents or results that they get from the mongo shell - these files would have names like the latter. This change would imply that the upload button might not show up for them. 

I haven't played with the extension incorporating your changes, so I could be wrong here.

**Comment by tec-goblin** [comment]
If I understand correctly the issue you describe, this is already not working. If I "save as" a cosmos-document.json and reopen Visual Studio Code, the upload button is there but it fails ("convertFromString of undefined"). That's because the saved document has lost its connection with the Azure tree, so the plugin doesn't know to which collection to save the file.

**Comment by EricJizbaMSFT** [comment]
FYI this PR was created before our recent switch from Travis CI to Azure Dev Ops. Push a commit (merging with master or addressing comments) and the "checks" for this PR will be updated

**Comment by PrashanthCorp** [comment]
@EricJizbaMSFT pointed out that you're saving with a "Save As" - we don't support persistence with that operation. Does the upload work fine with your changes when you use "Save" ?

**Comment by tec-goblin** [comment]
I am sorry for my insistence, I wasn't even aware of that issue and I tried to keep tab names short. The new commit should have fixed the issue

File: src/extension.ts [hunk 1]
```diff
@@ -122,10 +122,11 @@ export function activate(context: vscode.ExtensionContext) {
 			node = <MongoDocumentTreeItem | DocDBDocumentTreeItem>await tree.showTreeItemPicker([MongoDocumentTreeItem.contextValue, DocDBDocumentTreeItem.contextValue]);
 		}
 
+		const editorTabName = node.label + "cosmos-document.json";
 		if (node instanceof MongoDocumentTreeItem) {
-			await editorManager.showDocument(new MongoDocumentNodeEditor(node), 'cosmos-document.json');
+			await editorManager.showDocument(new MongoDocumentNodeEditor(node), editorTabName);
 		} else {
-			await editorManager.showDocument(new DocDBDocumentNodeEditor(node), 'cosmos-document.json');
+			await editorManager.showDocument(new DocDBDocumentNodeEditor(node), editorTabName);
```
[the remark this answers is not in the document]
Comment (tec-goblin) [reply src/extension.ts]: Sorry I missed that! 
I also added an "-" before the new name to make it a bit more readable.

**Pull request merged**